server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -168,12 +171,20 @@ public void processFlushEvent(ShuffleDataFlushEvent event) throws Exception {
               storageDataReplica,
               user,
               maxConcurrencyPerPartitionToWrite);
-      ShuffleWriteHandler handler = storage.getOrCreateWriteHandler(request);
-      long startTime = System.currentTimeMillis();
-      boolean writeSuccess = storageManager.write(storage, handler, event);
-      if (!writeSuccess) {
-        throw new EventRetryException();
+
+      long startTime = 0L;
+      try {
+        ShuffleWriteHandler handler = storage.getOrCreateWriteHandler(request);
+        startTime = System.currentTimeMillis();
+        boolean writeSuccess = storageManager.write(storage, handler, event);
+        if (!writeSuccess) {
+          shuffleIdsWithWriteError.add(event.getShuffleId());
```

Comment:
I don't think this is thread-safe.

It could be overwritten by other threads, which may be successful.

You need to consider concurrency problems through this PR.

Reply:
Apologies for the delayed response, but here I used `CopyOnWriteArrayList` for implementing `shuffleIdsWithWriteError`, do you still think thread safety is a concern here, or do you think the other parts of the code are not thread safe?

Reply:
Please help to review this.Thx.@maobaolong